src/couch_views/src/couch_views_indexer.erl

```diff
@@ -87,7 +89,52 @@ init() ->
         design_opts => Mrst#mrst.design_opts
     },
 
-    update(Db, Mrst, State).
+    try
+        update(Db, Mrst, State)
+    catch
+        exit:normal ->
+            ok;
+        Error:Reason  ->
+            NewRetry = Retries + 1,
+            RetryLimit = retry_limit(),
+
+            case should_retry(NewRetry, RetryLimit, Error) of
```

Comment:
Here we'd want to send the `Reason` value to `should_retry` as Error will be just exit|throw|error atoms.

Reply:
good spot on this